src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,100 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.exception.DukeInvalidDateException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the file used to store the task list.
+ */
+public class Storage {
+
+    private File file;
+
+    /**
+     * Creates a file object, and creating the necessary file or directory if file does not exist.
+     *
+     * @param filePath the file path of the file that stores the task list.
+     */
+    public Storage(String filePath) {
+        String[] path = filePath.split("/");
+        String root = Paths.get("").toAbsolutePath().toString();
+        this.file = new File(filePath);
+        if (!file.exists()) {
+            try {
+                Files.createDirectories(Paths.get(root + File.separator + path[0]));
+                Files.createFile(Paths.get(root + File.separator + path[0] + File.separator + path[1]));
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
You have done a good job with custom exceptions, would have liked to see that here.
In general, printing stack trace is not considered good exception handling. Think what better you can do here when an exception occurs?